notebooks/predicting_queries/simple_intent_suggestion.ipynb

```diff
@@ -0,0 +1,570 @@
+    "title_df = pd.read_csv(PROJECT_DIR+'/data/processed/mainstreamcontent.csv')\n",
+    "title_df = title_df.dropna()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#We want to weight titles a bit highly, so err just multiply by 3\n",
+    "title_df['body_tokens'] = title_df['text'] + ' ' + (title_df['title'] + ' ' * 3)\n",
+    "title_df['body_tokens'] = title_df['body_tokens'].map(lambda x: tokenizer.tokenize(x.translate(translator).split()))\n",
+    "title_df['body_tokens'] = title_df['body_tokens'].map(lambda x: [token.lower() for token in x])\n",
+    "title_df = title_df.dropna()\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "title_df.head(1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Intersection of words in queries + content\n",
+    "service_terms= set(title_df['body_tokens'].explode('body_tokens'))\n",
+    "# Lose some stop words\n",
+    "intersection = service_terms.difference(set(['i','of','the','a','and','to','you','if','that','then','in','on']))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "title_df['intersect_tokens'] = title_df['body_tokens'].map(lambda x: [token for token in x if token in intersection])\n",
+    "title_df['body_token_count'] = title_df['intersect_tokens'].map(lambda x: Counter(x))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# bag of words for mainstream content\n",
+    "service_bow = pd.DataFrame.from_records(title_df['body_token_count'],index=title_df['name'])\n",
+    "service_bow = service_bow.replace(np.nan,0)\n",
+    "#add one smoothing\n",
+    "service_bow = service_bow +1\n",
+    "service_dis = service_bow.divide(service_bow.sum(1),0)\n",
+    "service_dis.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query_taxons = query_taxons[['taxon','tokens']]\n",
+    "query_taxons['valid_tokens'] = query_taxons['tokens'].map(lambda x: [token for token in x if token in intersection])\n",
+    "query_taxons['valid_token_count'] = query_taxons['valid_tokens'].map(lambda x:Counter(x))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Bag of words for taxon queries\n",
+    "query_bow = pd.DataFrame.from_records(query_taxons['valid_token_count'],index=query_taxons['taxon'],columns=service_dis.columns)\n",
+    "# Don't smooth unless you have lots of data\n",
+    "query_bow = query_bow.replace(np.nan,0)\n",
+    "\n",
+    "query_dis = query_bow.divide(query_bow.sum(1),0)\n",
```

Comment:
I used this method here to get normalised BoW. Should be equivalent as you're doing it from first principles. 

https://github.com/alphagov/govuk-entity-personalisation/blob/main/notebooks/entities-synonyms/01_bow_tfidf_embeddings.py#L18

Reply:
added a check to make sure its working as intended